### DISCUSSION TITLE
Whats the best way to add custom reactions to each post in a custom app view?

### DISCUSSION THREAD

**Discussion by viksit** [bluesky-social/atproto#1285] [How to do it?] [answered]
Does bsky.app itself support it via xrpc/lexicons, or would it necessitate running a custom atproto server, and then add on support via a custom lexicon, and write a fallback for it for other clients?

Love what you're building!

**Comment by bnewbold** [comment, marked as answer]
Good question! There are likely multiple ways this could be accomplished, and we don't yet have enough experience with feature extensions to existing Lexicons to recommend a best practice. Creating an entirely new application/modality would be relatively more straight-forward.

One way this could work:

- create a new record type under a separate namespace, eg, `io.viksit.feed.reaction`. the schema would be similar to `app.bsky.feed.like`, but would include an emoji field (or string, or whatever you are thinking)
- implement a small AppView which subscribes to the global BGS firehose, filters to only records of this type, and maintains a database of reactions by subject (AT-URI). could be a SQL database, or whatever you are familiar with
- create a new app, or fork an existing app, to allow both creating these reaction records, and doing additional hydration queries to your AppView (proxied through the PDS) for display. Would probably want to have an API (Lexicon endpoint) that takes an array of AT-URIs, and returns, for each, a set of top reactions, and counts. possibly also a way to list the accounts which made each reaction, for an inspect view
- over time, if your extension gets traction, other client apps could add support

In theory, you could also advocate to the existing app namespace authority (for `app.bsky.*`, that would be us, Bluesky) to add this feature. But I think you are asking how to do an independent extension, which is more interesting and unique to how atproto works!
